### PR TITLE
v4l2tools: fix missing dependency

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Should fix errors in the form of:

    Package v4l2tools is missing dependencies for the following libraries: libvpx.so.6

Signed-off-by: Nick Hainke <vincent@systemli.org>

Maintainer: @mpromonet
Compile tested: tbd
Run tested: tbd

### CODE REVIEW

**Comment by PolynomialDivision** [comment]
I did not runtest only compile test.

**Comment by mpromonet** [comment]
Hi,

The project v4l2tools can compile with vpx or not (it is the same with x264/x265). The build problem seems CMake detect libvpx is available (using pkg-tools) but openwrt build not have in dependency. It may be nice to have a conditional dependency ?

Best Regards,
Michel.

**Comment by PolynomialDivision** [comment]
Any suggestion how to do this?

**Comment by PolynomialDivision** [comment]
@mpromonet Just to be sure, you mean that libvpx is available from the host?

**Comment by BKPepe** [comment]
@PolynomialDivision Please, don't blindly add more dependencies to packages when they are autodetected to just try to fix their compilation. They can be disabled. If you add a dependency, then package sizes are going to be increased. It could somehow change the current behavior of the package, and users don't expect it. You need to verify if it works. 

TL;DR: Let's imagine that you added a new dependency that is autodetected. You need to consider ``pros`` and ``cons`` to add it or disable it. Sometimes it can be handy, sometimes not. If it is added, you need to verify that the package is working, but you need to check also if the new feature, which depends on the package you added, works. Sometimes, it does not worth the hassle, and it is good to disable it. Usually, you don't need all the features which the package offers.

**Comment by PolynomialDivision** [comment]
Yeah sorry. We had this update of ipq40xx to 5.10er kernel. And I wanted to go through fail logs and look whats wrong and fails. Now I'm realizing I dig out already existing failures.

**Review by mpromonet — COMMENTED** [review]
It may be better to use : 
`	DEPENDS:=+libstdcpp +libjpeg +PACKAGE_libvpx:libvpx`

**Comment by stintel** [comment]
> It may be better to use :
> ` DEPENDS:=+libstdcpp +libjpeg +PACKAGE_libvpx:libvpx`

The best way is to add an option in the v4l2tools build to disable libvpx, and use that in the OpenWrt Makefile. Like this, packages build by the buildbots, and thus the packages in the repo, will still have the dependency.
